src/Components/VideoActions/VideoActions.jsx

```diff
@@ -2,13 +2,167 @@ import React, { useEffect, useState } from "react";
 import "./VideoActions.css";
 import { useAuth } from "../../Context/Auth-Context";
 import { useUserData } from "../../Context/UserData-Context";
+import toast, { Toaster } from "react-hot-toast";
 
 import axios from "axios";
 const VideoActions = (props) => {
-  
+  const {
+    currentIdHandler,
+    modalVisibleHandler,
+    currentVideoHandler,
+    sortByHandler,
+  } = useUserData();
+  const { authToken } = useAuth();
+  const [likedVideos, setLikedVideos] = useState([]);
+  const [likedRender, setLikedRender] = useState(false);
+  const [watchLaterVideos, setWatchLaterVideos] = useState([]);
+  const [linkText, setLinkText] = useState("Copy Link");
+  // Get Liked Videos from the backend
+  useEffect(() => {
+    (async () => {
+      const response = await axios.get("/api/user/likes", {
+        headers: {
+          authorization: authToken,
+        },
+      });
+      setLikedVideos(response.data.likes);
+    })();
+  }, [likedRender]);
+
+  // Get Watch Later Videos from the backend
+  useEffect(() => {
+    (async () => {
+      const response = await axios.get("/api/user/watchLater", {
+        headers: {
+          authorization: JSON.parse(localStorage.getItem("user")).encodedToken,
+        },
+      });
+      setWatchLaterVideos(response.data.watchLater);
+    })();
+  }, []);
+
+  const getWatchLaterVideos = async () => {
+    const response = await axios.get("/api/user/watchLater", {
+      headers: {
+        authorization: JSON.parse(localStorage.getItem("user")).encodedToken,
+      },
+    });
+    setWatchLaterVideos(response.data.watchLater);
+  };
+
+  //Add to Watch Later Videos
+  const watchLaterHandler = async () => {
+    const response = await axios.post(
+      "/api/user/watchLater",
+      {
+        video: props.video,
+      },
+      {
+        headers: {
+          authorization: JSON.parse(localStorage.getItem("user")).encodedToken,
+        },
+      }
+    );
+    getWatchLaterVideos();
+  };
+
+  //Remove from Watch Later videos
+  const removeFromWatchLaterHandler = async () => {
+    const response = await axios.delete(
+      `/api/user/watchLater/${props.video._id}`,
+      {
+        headers: {
+          authorization: JSON.parse(localStorage.getItem("user")).encodedToken,
+        },
```

Comment:
You can extract localStorage as separate function

Reply:
or extract the encoded token from localStorage and store it in the state.

Reply:
I did that but it wasn't updating the state correctly for some reason so I decided to hard code it everywhere. Will check that thank you!